### PR TITLE
Cyb 50 weclome dashboard

### PR DESCRIPTION
**Pull request by Ganoush69**
Dashboard, sidebar, topbar, hopes and dreams.

### CODE REVIEW

File: index.html [hunk 1]
```diff
@@ -8,6 +8,7 @@
     <title>Vite + React + TS</title>
   </head>
   <body>
+    <noscript>You need to enable javascript to progress further.</noscript>
```
Comment (JaegerCryptic) [inline index.html:11]: Not necessary but I like it

File: package.json [hunk 2]
```diff
@@ -17,6 +17,9 @@
         "@emotion/react": "^11.11.4",
         "@emotion/styled": "^11.11.0",
         "@fontsource/roboto": "^5.0.12",
+        "@fortawesome/fontawesome-svg-core": "^6.6.0",
+        "@fortawesome/free-solid-svg-icons": "^6.6.0",
+        "@fortawesome/react-fontawesome": "^0.2.2",
```
Comment (JaegerCryptic) [inline package.json:22]: What are these used for?

File: src/common/components/header/header.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react';
+import '../../../theme/topbar.css';
```
Comment (JaegerCryptic) [inline src/common/components/header/header.tsx]: My fault. 
When we have internal imports and third party imports we put the third party at the top and seperate them from the internal by adding a space
```suggestion
import React, { useState } from 'react';

import '../../../theme/topbar.css';
```

File: src/features/dashboard/dashboard.tsx [hunk 4]
```diff
@@ -0,0 +1,24 @@
+import  Sidebar  from '../../common/components/sidebar/sidebar.tsx' // This is importing the sidebar that i am currently trying to work on.
+import  TopBar  from '../../common/components/header/header.tsx' // same story with the header/topbar 
+import  React from 'react'
```
Comment (JaegerCryptic) [inline src/features/dashboard/dashboard.tsx:3]: imports

File: src/features/startup/SignIn.tsx [hunk 5]
```diff
@@ -4,13 +4,15 @@ import { useQueryClient } from "@tanstack/react-query";
 import { Icon, Typography } from "@mui/material";
 import Grid from "@mui/material/Unstable_Grid2/Grid2";
 import { z } from "zod";
-
+import { Route } from 'react-router-dom';
```
Comment (JaegerCryptic) [inline src/features/startup/SignIn.tsx:7]: imports

File: src/common/components/header/header.tsx [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+            <p>SCP Foundation 2</p>
+            <p>SCP Foundation 3</p>
+          </div>
+        )}
+      </div>
+    </div>
+  );
+};
+
+export default TopBar;
```
Comment (JaegerCryptic) [inline src/common/components/header/header.tsx]:
```suggestion
```

File: src/common/components/header/header.tsx [hunk 7]
```diff
@@ -0,0 +1,31 @@
+import React, { useState } from 'react';
+import '../../../theme/topbar.css';
+
+const TopBar = () => {
```
Comment (JaegerCryptic) [inline src/common/components/header/header.tsx]:
```suggestion
export const TopBar = () => {
```

File: src/common/components/header/header.tsx [hunk 8]
```diff
@@ -0,0 +1,31 @@
...
+          <div className="dropdown-menu">
+            <p>SCP Foundation 1</p>
+            <p>SCP Foundation 2</p>
+            <p>SCP Foundation 3</p>
+          </div>
+        )}
+      </div>
+    </div>
+  );
+};
```
Comment (JaegerCryptic) [inline src/common/components/header/header.tsx:30]: Careful with using html elements directly like this. If its good atm that's fine but we have mui components already in the project like CLButton for this

File: src/common/components/sidebar/sidebar.tsx [hunk 9]
```diff
@@ -0,0 +1,49 @@
...
+              <li>Thing 4</li>
+            </ul>
+          )}
+          <li>Menu Item 2</li>
+          <li>Menu Item 3</li>
+        </ul>
+      </nav>
+    </div>
+  );
+};
```
Comment (JaegerCryptic) [inline src/common/components/sidebar/sidebar.tsx:47]: Same as above

File: src/common/components/sidebar/sidebar.tsx [hunk 10]
```diff
@@ -0,0 +1,49 @@
...
+
+export default Sidebar;
```
Comment (JaegerCryptic) [inline src/common/components/sidebar/sidebar.tsx]:
```suggestion
```

File: src/common/components/sidebar/sidebar.tsx [hunk 11]
```diff
@@ -0,0 +1,49 @@
+import React, { useState } from 'react';
+import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
+import { faBars, faChevronDown } from '@fortawesome/free-solid-svg-icons';
+import '../../../theme/sidebar.css'
+
+// sidebar time
+const Sidebar = () => {
```
Comment (JaegerCryptic) [inline src/common/components/sidebar/sidebar.tsx]:
```suggestion
export const Sidebar = () => {
```

File: src/features/dashboard/dashboard.tsx [hunk 12]
```diff
@@ -0,0 +1,24 @@
+import  Sidebar  from '../../common/components/sidebar/sidebar.tsx' // This is importing the sidebar that i am currently trying to work on.
```
Comment (JaegerCryptic) [inline src/features/dashboard/dashboard.tsx:1]: import

File: src/routes/Root.tsx [hunk 13]
```diff
@@ -1,4 +1,5 @@
 import { Navigate, RouteObject, useLocation } from 'react-router-dom'
+import  Dashboard from '../features/dashboard/dashboard.tsx'
```
Comment (JaegerCryptic) [inline src/routes/Root.tsx]: import

File: src/theme/sidebar.css [hunk 14]
```diff
@@ -0,0 +1,47 @@
...
+  }
+  
+  ul li:hover {
+    background-color: #444;
+  }
+  
+  .dropdown {
+    background-color: #444;
+    padding-left: 20px;
+  }
```
Comment (JaegerCryptic) [inline src/theme/sidebar.css:47]: My fault for not being clear, but we don't really want to be having css files. We can create side bars and dropdowns with mui and then apply css inline without needed a css file

File: src/routes/Root.tsx [hunk 15]
```diff
@@ -14,3 +15,10 @@ export const ROOT_ROUTE: RouteObject = {
   element: <Root />,
   children: [],
 }
+
+// trying to define the dashboard as a proper route ... but im not sure this is the correct way to do it.
```
Comment (JaegerCryptic) [inline src/routes/Root.tsx]: You can remove all your route stuff. I'll fix it up lol

**Review by JaegerCryptic — CHANGES_REQUESTED** [review]

File: package.json (hunk 2 above)
Comment (Ganoush69) [reply package.json:22]: they were used in a previous code i wrote and referenced online

File: src/common/components/header/header.tsx (hunk 8 above)
Comment (Ganoush69) [reply src/common/components/header/header.tsx:30]: ill keep this in mind

File: src/routes/Root.tsx (hunk 15 above)
Comment (Ganoush69) [reply src/routes/Root.tsx]: thank you

**Review by JaegerCryptic — APPROVED** [review]

**Pull request merged**